src/compile.rs

```diff
@@ -18,46 +26,106 @@ pub fn compile_host(host: &Host) -> String {
 fn _compile_host(host: &Host) -> Result<String, fmt::Error> {
   let mut code = String::default();
 
-  let defs = host
-    .defs
-    .iter()
-    .filter_map(|(name, def)| Some((name, def.downcast_ref::<HostedDef<InterpretedDef>>()?)))
-    .map(|(raw_name, def)| (raw_name, sanitize_name(raw_name), def));
+  let mut def_infos: BTreeMap<&str, DefInfo<'_>> = BTreeMap::new();
+  for (hvmc_name, def) in &host.defs {
+    if let Some(def) = def.downcast_ref::<HostedDef<InterpretedDef>>() {
+      def_infos.insert(hvmc_name, DefInfo {
+        rust_name: sanitize_name(hvmc_name),
+        refs: refs(host, &def.data.0.instr),
+        def,
+      });
+    }
+  }
 
-  writeln!(code, "#![allow(non_upper_case_globals, unused_imports)]")?;
-  writeln!(code, "use crate::{{host::{{Host, DefRef}}, run::*, ops::{{TypedOp, Ty::*, Op::*}}}};")?;
+  writeln!(code, "#![allow(non_camel_case_types, unused_imports, unused_variables)]")?;
```

Comment:
hmm, might need a `non_snake_case_idents` or something here for if people write `@Foo`, but maybe we should just simplify to:
```suggestion
  writeln!(code, "#![allow(warnings)]")?;
```

Reply:
good idea